### PR TITLE
Add advertisedAddress config field to ProxyConfiguration

### PR DESCRIPTION
**Pull request by aahmed-se**
Proxy can have it's advertisedAddress configured like pulsar broker

### CODE REVIEW

File: pulsar-proxy/pom.xml [hunk 1]
```diff
@@ -165,5 +165,20 @@
       <groupId>com.beust</groupId>
       <artifactId>jcommander</artifactId>
     </dependency>
+
+    <dependency>
```
Comment (merlimat) [inline pulsar-proxy/pom.xml:169]: Are these new deps needed?

Comment (aahmed-se) [reply pulsar-proxy/pom.xml]: removed it they are required for running it in the ide only.

**Comment by wolfstudy** [comment]
Move this change to 2.6.2, because #8067 depends on it.

**Pull request merged**